src/codegen.cpp

```diff
@@ -5629,90 +5629,122 @@ static std::unique_ptr<Module> emit_function(
     DIFile *topfile = NULL;
     DISubprogram *SP = NULL;
     DebugLoc noDbg, topdebugloc;
-    if (ctx.debug_enabled) {
-        // TODO: Fix when moving to new LLVM version
-        topfile = dbuilder.createFile(ctx.file, ".");
-        DICompileUnit *CU = dbuilder.createCompileUnit(0x01, topfile, "julia", true, "", 0);
-        DISubroutineType *subrty;
-        if (jl_options.debug_level <= 1) {
-            subrty = jl_di_func_null_sig;
-        }
-        else if (!specsig) {
-            subrty = jl_di_func_sig;
-        }
-        else {
-            subrty = get_specsig_di(jlrettype, lam->specTypes, topfile, dbuilder);
-        }
-        SP = dbuilder.createFunction(CU
-                                     ,dbgFuncName      // Name
-                                     ,f->getName()     // LinkageName
-                                     ,topfile          // File
-                                     ,toplineno        // LineNo
-                                     ,subrty           // Ty
+    DICompileUnit::DebugEmissionKind debugKind;
+    switch (jl_options.debug_level) {
+        case 0:
+            debugKind = DICompileUnit::DebugEmissionKind::NoDebug;
+            break;
+        case 1:
+#if JL_LLVM_VERSION >= 80000
+            debugKind = DICompileUnit::DebugEmissionKind::DebugDirectivesOnly;
+            break;
+#endif
+        case 2:
+        default:
+            debugKind = DICompileUnit::DebugEmissionKind::FullDebug;
+            break;
+    }
+
+    // TODO: Fix when moving to new LLVM version
```

Comment:
@Keno  do you remember what this TODO is about? You added it in 2012 30aed865b713a546d2d2eb44b88b43aa1c27e1ed

Reply:
Yes, i think it was about switching the identifier from C89 to Julia